### PR TITLE
Add tests for ClearCommand, EchoCommand and HelpCommand

### PR DESCRIPTION
**Pull request by tlmii**

### CODE REVIEW

File: src/Microsoft.HttpRepl.IntegrationTests/Commands/ClearCommandTests.cs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+using Microsoft.Repl.ConsoleHandling;
+using Microsoft.Repl.Parsing;
+using Xunit;
+
+namespace Microsoft.HttpRepl.IntegrationTests.Commands
+{
+    public class ClearCommandTests : CommandTestsBase
+    {
+        [Fact]
+        public void CanHandle_Name_ReturnsTrue()
```
Comment (bhsubra) [inline src/Microsoft.HttpRepl.IntegrationTests/Commands/ClearCommandTests.cs]: Looks like happy paths are covered. Should we add tests for invalid inputs?

File: src/Microsoft.HttpRepl.IntegrationTests/Commands/EchoCommandTests.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.HttpRepl.Commands;
+using Microsoft.HttpRepl.IntegrationTests.Mocks;
+using Microsoft.Repl.Parsing;
+using Xunit;
+
+namespace Microsoft.HttpRepl.IntegrationTests.Commands
+{
+    public class EchoCommandTests : CommandTestsBase
```
Comment (bhsubra) [inline src/Microsoft.HttpRepl.IntegrationTests/Commands/EchoCommandTests.cs]: I am wondering if we should add tests for ExecuteAsync(..). Looks like it doesn't do much. But we could verify the EchoRequest value. I'll let you decide if its worth adding or not.

File: src/Microsoft.HttpRepl.IntegrationTests/Mocks/ConsoleManagerStub.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+        public bool IsKeyAvailable => default;
+
+        public bool AllowOutputRedirection => default;
+
+        public bool IsCaretVisible { get => default; set { } }
+
+        public IDisposable AddBreakHandler(Action onBreak) => default;
+
+
```
Comment (bhsubra) [inline src/Microsoft.HttpRepl.IntegrationTests/Mocks/ConsoleManagerStub.cs]: nit: empty line

File: src/Microsoft.HttpRepl.IntegrationTests/Mocks/ConsoleManagerStub.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
+using System;
```
Comment (bhsubra) [inline src/Microsoft.HttpRepl.IntegrationTests/Mocks/ConsoleManagerStub.cs]: nit: Missing copyrignt comment

File: src/Microsoft.HttpRepl.IntegrationTests/Commands/ClearCommandTests.cs [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+        {
+            ArrangeInputs("clear", out MockedShellState shellState, out HttpState httpState, out ICoreParseResult parseResult);
```
Comment (bhsubra) [inline src/Microsoft.HttpRepl.IntegrationTests/Commands/ClearCommandTests.cs]: nit: can we specify the parameter here for readability?

File: src/Microsoft.HttpRepl.IntegrationTests/Commands/ClearCommandTests.cs (hunk 1 above)
Comment (tlmii) [reply src/Microsoft.HttpRepl.IntegrationTests/Commands/ClearCommandTests.cs]: Added a couple of likely invalid inputs in https://github.com/aspnet/HttpRepl/pull/74/commits/af55ecc8092914f16ced3d5fd83db4b96b4de1de

File: src/Microsoft.HttpRepl.IntegrationTests/Mocks/ConsoleManagerStub.cs (hunk 4 above)
Comment (tlmii) [reply src/Microsoft.HttpRepl.IntegrationTests/Mocks/ConsoleManagerStub.cs]: Fixed in https://github.com/aspnet/HttpRepl/pull/74/commits/af55ecc8092914f16ced3d5fd83db4b96b4de1de

File: src/Microsoft.HttpRepl.IntegrationTests/Commands/ClearCommandTests.cs (hunk 5 above)
Comment (tlmii) [reply src/Microsoft.HttpRepl.IntegrationTests/Commands/ClearCommandTests.cs]: Fixed in https://github.com/aspnet/HttpRepl/pull/74/commits/af55ecc8092914f16ced3d5fd83db4b96b4de1de

File: src/Microsoft.HttpRepl.IntegrationTests/Mocks/ConsoleManagerStub.cs (hunk 3 above)
Comment (tlmii) [reply src/Microsoft.HttpRepl.IntegrationTests/Mocks/ConsoleManagerStub.cs]: Fixed in https://github.com/aspnet/HttpRepl/pull/74/commits/af55ecc8092914f16ced3d5fd83db4b96b4de1de

**Comment by tlmii** [comment]
The original PR build failed for some reason to do with git credentials. I kicked off [another one that succeeded](https://dev.azure.com/dnceng/public/_build/results?buildId=262916&view=logs&j=37bec900-c686-5858-5c69-ce96aeb4a5fe) but for some reason the PR isn't updating it to show that. Going to go ahead and merge regardless.

**Pull request merged**